contracts/token/ERC1155/ERC1155.sol

```diff
@@ -0,0 +1,234 @@
+pragma solidity ^0.5.0;
+
+import "./IERC1155.sol";
+import "./IERC1155TokenReceiver.sol";
+import "../../math/SafeMath.sol";
+import "../../utils/Address.sol";
+import "../../introspection/ERC165.sol";
+
+/**
+ * @title Standard ERC1155 token
+ *
+ * @dev Implementation of the basic standard multi-token.
+ * See https://eips.ethereum.org/EIPS/eip-1155
+ * Originally based on code by Enjin: https://github.com/enjin/erc-1155
+ */
+contract ERC1155 is ERC165, IERC1155
```

Comment:
In these contracts there isn't a `totalSupply` variable like in other assets (e.g. `ERC20`). We know that the EIP says '_The total value transferred from address `0x0` minus the total value transferred to `0x0` observed via the `TransferSingle` and `TransferBatch` events MAY be used by clients and exchanges to determine the "circulating supply" for a given token ID_' and having a `totalSupply` is not necessary, but is that expected to be implemented in another derived contract? Or it would only rely on the offchain systems?

Reply:
I tried to write as little code as I could to arrive at a compliant implementation. :man_shrugging: 

In any case, EIP 20 mentions `totalSupply`, but not EIP 1155. In fact, IIRC, there was a point in the EIP 1155 discussion when `totalSupply` referred to the total supply across _all token IDs_ and accounts, and would be ERC20-like. Since that name's taken for ERC-20, I'd err on the side of caution and not include some non-standard interpretation.

Also, adding a `totalSupply` tracker by ID would _technically_ mean contracts must not exceed the max word in supply for any particular ID across all addresses.

I feel like the proper way to add this feature would be to change the name maybe to `totalSupplyOfId` or something, and to have it in a derived extension, maybe with some EIP backing it. I don't think it's in scope for this PR though.

Reply:
`totalSupply(uint256 id)` might make sense, but I agree in that it is out of scope for this